### ISSUE TITLE
Feature request: isolating extrinsic dryrun changes from block hook changes

### ISSUE THREAD

**Issue by sander2** [AcalaNetwork/chopsticks#295]
Currently, the output of the `dry-run` command is pretty noisy, since it includes changes caused be on_initialize and on_finalize hooks. It would be super helpful if there would be a way to only display the changes caused by the execution of the extrinsic itself. I'm not sure if this is feasible to implement. Maybe an approximation could be made by forking the chain and mining two blocks on different forks, and then comparing the result of both

**Comment by sander2** [comment]
Yea I was using that. It just has a lot of noise. For dev work that is not a problem. However, we're starting to use dry-run to demonstrate the effects of democracy proposals to other people, and then it helps a lot if the changes are as concise as possible, without irrelevant storage changes caused by block hooks

**Comment by xlc** [comment]
Yeah we can exclude results from hooks. Just need some refactoring.
